utoipauto-core/src/attribute_utils.rs

```diff
@@ -27,257 +36,276 @@ pub fn update_openapi_macro_attributes(
-    let src_uto_macro = remove_responses(src_uto_macro);
-    let src_uto_macro = remove_components(src_uto_macro);
-
-    let paths = format!("{}{}", uto_paths, paths);
-    let schemas = format!("{}{}", uto_models, schemas);
-    let responses = format!("{}{}", uto_responses, responses);
-    let src_uto_macro = format!(
-        "paths({}),components(schemas({}),responses({})),{}",
-        paths, schemas, responses, src_uto_macro
-    )
-    .replace(",,", ",");
+    let paths = extract_paths(&nested_attributes);
+    let schemas = extract_schemas(&nested_attributes);
+    let responses = extract_responses(&nested_attributes);
+    let remaining_nested_attributes = remove_paths_and_components(nested_attributes);
 
-    let stream: proc_macro2::TokenStream = src_uto_macro.parse().unwrap();
-
-    syn::parse_quote! { #[openapi( #stream )] }
-}
+    let uto_paths = match uto_paths.is_empty() {
+        true => TokenStream::new(),
+        false => quote::quote!(#uto_paths,),
+    };
+    let uto_models = match uto_models.is_empty() {
+        true => TokenStream::new(),
+        false => quote::quote!(#uto_models,),
+    };
+    let uto_responses = match uto_responses.is_empty() {
+        true => TokenStream::new(),
+        false => quote::quote!(#uto_responses,),
+    };
+    let uto_macro = quote::quote!(
+        paths(#uto_paths #paths),components(schemas(#uto_models #schemas),responses(#uto_responses #responses)),
+        #remaining_nested_attributes
+    );
 
-fn remove_paths(src_uto_macro: String) -> String {
-    if src_uto_macro.contains("paths(") {
-        let paths = src_uto_macro.split("paths(").collect::<Vec<&str>>()[1];
-        let paths = paths.split(')').collect::<Vec<&str>>()[0];
-        src_uto_macro
-            .replace(format!("paths({})", paths).as_str(), "")
-            .replace(",,", ",")
-    } else {
-        src_uto_macro
-    }
-}
-
-fn remove_schemas(src_uto_macro: String) -> String {
-    if src_uto_macro.contains("schemas(") {
-        let schemas = src_uto_macro.split("schemas(").collect::<Vec<&str>>()[1];
-        let schemas = schemas.split(')').collect::<Vec<&str>>()[0];
-        src_uto_macro
-            .replace(format!("schemas({})", schemas).as_str(), "")
-            .replace(",,", ",")
-    } else {
-        src_uto_macro
-    }
+    syn::parse_quote! { #[openapi( #uto_macro )] }
 }
 
-fn remove_components(src_uto_macro: String) -> String {
-    if src_uto_macro.contains("components(") {
-        let components = src_uto_macro.split("components(").collect::<Vec<&str>>()[1];
-        let components = components.split(')').collect::<Vec<&str>>()[0];
-        src_uto_macro
-            .replace(format!("components({})", components).as_str(), "")
-            .replace(",,", ",")
-    } else {
-        src_uto_macro
+fn remove_paths_and_components(nested_attributes: Punctuated<Meta, Token![,]>) -> TokenStream {
+    let mut remaining = Vec::with_capacity(nested_attributes.len());
```

Comment:
```suggestion
    let mut remaining = Vec::new();
```
As we only add `responses` more memory then needed is being allocated.
We can take the small performance hit here.

Reply:
You're right, I think I changed this before changing the function body and then it stuck. Reverted it back to just `Vec::new` with the last commit.